pyquil/quil.py

```diff
@@ -40,6 +40,9 @@ def synthesize(self, resource_manager=None):
         self.resource_manager.reset()
         return super(Program, self).synthesize(resource_manager)
 
+    def __eq__(self, other):
```

Comment:
yeah I don't want to get into the full use case of equality as there's a lot to check if you are thinking about really checking "functional" equivalent of pyquil program objects. generated labels and qubits etc can come up.  The pseudo-code that you've added is a little funky as there is hidden state that is being changed and relied upon. I think I'm instead just going to switch to using `.out()` for the comparison in the test as that is actually what I need for my test use case.